testutils/call.go

```diff
@@ -158,6 +161,22 @@ func (f *FakeCallFrame) Arg2Iterator() (arg2.KeyValIterator, error) {
 	return f.arg2KVIterator, f.hasArg2KVIterator
 }
 
+// Arg2Append appends a key value pair to Arg2
+func (f *FakeCallFrame) Arg2Append(key, val []byte) {
+	kv := make(map[string]string)
+	for iter, err := f.Arg2Iterator(); err == nil; iter, err = iter.Next() {
+		kv[string(iter.Key())] = string(iter.Value())
+	}
+	kv[string(key)] = string(val)
+
+	iterator, err := arg2.NewKeyValIterator(thriftarg2test.BuildKVBuffer(kv))
+	require.NoError(f.tb, err)
+
+	f.arg2KVIterator = iterator
+
+	return
+}
```

Comment:
i think we should just keep a list of appends, we don't want to really mutate arg2 (that is also not the behaviour of the real call frame)